crates/chain/src/indexed_tx_graph.rs

```diff
@@ -292,20 +297,24 @@ where
 
     /// Batch insert all transactions of the given `block` of `height`.
     ///
-    /// Each inserted transaction's anchor will be constructed from
-    /// [`AnchorFromBlockPosition::from_block_position`].
+    /// Each inserted transaction's anchor will be constructed using [`TxPosInBlock`].
     ///
     /// To only insert relevant transactions, use [`apply_block_relevant`] instead.
     ///
     /// [`apply_block_relevant`]: IndexedTxGraph::apply_block_relevant
-    pub fn apply_block(&mut self, block: Block, height: u32) -> ChangeSet<A, I::ChangeSet> {
+    pub fn apply_block(&mut self, block: &Block, height: u32) -> ChangeSet<A, I::ChangeSet> {
```

Comment:
Is there a specific reason for this change?

Reply:
Ah, not anymore. If I remember, I had to add it in the first iteration when the lifetime for the struct was placed on the impl. No need for it anymore.

I reverted to the original.